processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryQueryToolChestTest.java

```diff
@@ -929,6 +930,64 @@ private void doTestCacheStrategy(final ValueType valueType, final Object dimValu
     Assert.assertEquals(typeAdjustedResult2, fromResultCacheResult);
   }
 
+  @Test
+  public void testQueryCacheKeyWithLimitSpec()
+  {
+    final GroupByQuery query1 = GroupByQuery
```

Comment:
Would you please add another query which has the same spec and `CTX_KEY_APPLY_LIMIT_PUSH_DOWN` of true in the context, and check the equality of the cache key of the query and `query1`?

Reply:
Oh sorry, I forgot to approve this PR. This comment is not a blocker if you want to do in a follow-up.

Reply:
added test for this.